staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/connection.go

```diff
@@ -193,12 +193,12 @@ func (c *connection) sendPings(period time.Duration) {
 		case <-c.conn.CloseChan():
 			return
 		case <-t.C:
-		}
-		if _, err := c.ping(); err != nil {
-			klog.V(3).Infof("SPDY Ping failed: %v", err)
-			// Continue, in case this is a transient failure.
-			// c.conn.CloseChan above will tell us when the connection is
-			// actually closed.
+			if _, err := c.ping(); err != nil {
```

Comment:
can you recover this change and test again to understand what is the change that makes the difference? if this one or the atomic.Int64 one

Reply:
@aojea I've reverted this change back to how it was keeping only the atomic.Int64 one. Under stress the test has been run 100k times without any failure. IMHO this change does not make any difference, I moved the ping call directly under the tick case because I felt it was clearer. It's completely subjective. If you prefer the original way I will update the PR.

<img width="535" alt="Screenshot 2023-05-10 at 20 38 19" src="https://github.com/kubernetes/kubernetes/assets/9576370/385e9b2f-eeae-4105-895f-92a536e766bb">

Reply:
only one change at a time is easier to think about